helper/schema/resource_diff_test.go

```diff
@@ -1237,6 +1237,99 @@ func TestGetChangedKeysPrefix(t *testing.T) {
 				"foo",
 			},
 		},
+		{
+			Name: "basic primitive diff with empty prefix",
+			Schema: map[string]*Schema{
+				"foo": {
+					Type:     TypeString,
+					Optional: true,
+					Computed: true,
+				},
+				"qux": {
+					Type:     TypeString,
+					Optional: true,
+					Computed: true,
+				},
+			},
+			State: &terraform.InstanceState{
+				Attributes: map[string]string{
+					"foo": "bar",
+					"qux": "abc",
+				},
+			},
+			Config: testConfig(t, map[string]interface{}{
+				"foo": "baz",
+				"qux": "abc",
+			}),
+			Diff: &terraform.InstanceDiff{
+				Attributes: map[string]*terraform.ResourceAttrDiff{
+					"foo": {
+						Old: "bar",
+						New: "baz",
+					},
+				},
+			},
+			Key: "",
+			ExpectedKeys: []string{
+				"foo",
+			},
+		},
+		{
+			Name: "nested diff with empty prefix",
+			Schema: map[string]*Schema{
+				"foo": {
+					Type:     TypeString,
+					Optional: true,
+					Computed: true,
+				},
+				"bar": {
+					Type:     TypeList,
+					Required: true,
+					MaxItems: 1,
+					Elem: &Resource{
+						Schema: map[string]*Schema{
+							"baz": {
+								Type:     TypeString,
+								Optional: true,
+							},
+							"qux": {
+								Type:     TypeString,
+								Optional: true,
+							},
+						},
+					},
+				},
+			},
+			State: &terraform.InstanceState{
+				Attributes: map[string]string{
+					"foo":       "abc",
+					"bar.#":     "1",
+					"bar.0.baz": "def",
+					"bar.0.qux": "xyz",
+				},
+			},
+			Config: testConfig(t, map[string]interface{}{
+				"foo": "abc",
+				"bar": []interface{}{
+					map[string]interface{}{
+						"baz": "abc",
```

Comment:
The method relies on the diff, but this will make state == config, just in case that might confuse future travelers 👍 

```suggestion
						"baz": "def",
```